agent/engine/engine_sudo_linux_integ_test.go

```diff
@@ -127,3 +153,166 @@ func createTestLocalVolumeMountTask() *apitask.Task {
+	cfg.DataDirOnHost = "/var/lib/ecs/"
+	cfg.TaskCleanupWaitDuration = 1 * time.Second
+	taskEngine, done, _ := setup(cfg, nil, t)
+	defer done()
+
+	testTask := createLogDriverAndContainersUsingLogDriverTask(t)
+	taskEngine.(*DockerTaskEngine).resourceFields = &taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			EC2InstanceID: testInstanceID,
+		},
+	}
+	go taskEngine.AddTask(testTask)
+
+	testEvents := InitEventCollection(taskEngine)
+
+	//Verify logsender container is running
+	err := VerifyContainerStatus(apicontainerstatus.ContainerRunning, testTask.Arn+":logsender", testEvents, t)
+	assert.NoError(t, err, "Verify logsender is running")
+
+	//Verify logrouter container is running
+	err = VerifyContainerStatus(apicontainerstatus.ContainerRunning, testTask.Arn+":logrouter", testEvents, t)
+	assert.NoError(t, err, "Verify logrouter is running")
+
+	////Verify task is in running state
+	err = VerifyTaskStatus(apitaskstatus.TaskRunning, testTask.Arn, testEvents, t)
+	assert.NoError(t, err, "Not verified task running")
+
+	taskID, err := testTask.GetID()
+
+	//Manually stop logrouter container
+	cont0, ok := testTask.ContainerByName("logrouter")
+	taskEngine.(*DockerTaskEngine).stopContainer(testTask, cont0)
+
+	// declare a cloudwatch client
+	cwlClient := cloudwatchlogs.New(session.New(), aws.NewConfig().WithRegion(testECSRegion))
+	params := &cloudwatchlogs.GetLogEventsInput{
+		LogGroupName:  aws.String(testLogGroupName),
+		LogStreamName: aws.String(fmt.Sprintf("logrouter-fluentbit-logsender-%s", taskID)),
+	}
+
+	// wait for the cloud watch logs
+	resp, err := waitCloudwatchLogs(cwlClient, params)
+	require.NoError(t, err)
+	// there should only be one event as we are echoing only one thing that part of the include-filter
+	assert.Equal(t, 1, len(resp.Events))
+
+	message := aws.StringValue(resp.Events[0].Message)
+	jsonBlob := make(map[string]string)
+	err = json.Unmarshal([]byte(message), &jsonBlob)
+	require.NoError(t, err)
+	assert.Equal(t, "stdout", jsonBlob["source"])
+	assert.Equal(t, "include", jsonBlob["log"])
+	assert.Contains(t, jsonBlob, "container_id")
+	assert.Contains(t, jsonBlob["container_name"], "logsender")
+
+	//Verify logsender container is stopped
+	err = VerifyContainerStatus(apicontainerstatus.ContainerStopped, testTask.Arn+":logsender", testEvents, t)
+	assert.NoError(t, err)
+
+	//Verify logrouter container is stopped
+	err = VerifyContainerStatus(apicontainerstatus.ContainerStopped, testTask.Arn+":logrouter", testEvents, t)
+	assert.NoError(t, err)
+
+	//Verify the task itself has stopped
+	err = VerifyTaskStatus(apitaskstatus.TaskStopped, testTask.Arn, testEvents, t)
+	assert.NoError(t, err)
+
+	time.Sleep(cfg.TaskCleanupWaitDuration)
+	for i := 0; i < 60; i++ {
+		_, ok = taskEngine.(*DockerTaskEngine).State().TaskByArn(testTask.Arn)
+		if !ok {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	// Make sure all the resource is cleaned up
```

Comment:
can you check that the directory /var/lib/ecs/data/logrouter/<task-id> is removed as well?